frame/dapps-staking/src/lib.rs

```diff
@@ -0,0 +1,282 @@
+    /// Receive no reward.
+    None,
+}
+
+impl<AccountId> Default for RewardDestination<AccountId> {
+    fn default() -> Self {
+        RewardDestination::Staked
+    }
+}
+
+/// Information regarding the active era (era in used in session).
+#[derive(Encode, Decode, RuntimeDebug)]
+pub struct ActiveEraInfo {
+    /// Index of era.
+    pub index: EraIndex,
+    /// Moment of start expressed as millisecond from `$UNIX_EPOCH`.
+    ///
+    /// Start can be none if start hasn't been set for the era yet,
+    /// Start is set on the first on_finalize of the era to guarantee usage of `Time`.
+    start: Option<u64>,
+}
+
+/// Handler for determining how much of a balance should be paid out on the current era.
+pub trait EraPayout<Balance> {
+    /// Determine the payout for this era.
+    ///
+    /// Returns the amount to be paid to stakers in this era, as well as whatever else should be
+    /// paid out ("the rest").
+    fn era_payout(
+        total_staked: Balance,
+        total_issuance: Balance,
+        era_duration_millis: u64,
+    ) -> (Balance, Balance);
+}
+
+impl<Balance: Default> EraPayout<Balance> for () {
+    fn era_payout(
+        _total_staked: Balance,
+        _total_issuance: Balance,
+        _era_duration_millis: u64,
+    ) -> (Balance, Balance) {
+        (Default::default(), Default::default())
+    }
+}
+
+/// Mode of era-forcing.
+#[derive(Copy, Clone, PartialEq, Eq, Encode, Decode, RuntimeDebug)]
+#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
+pub enum Forcing {
+    /// Not forcing anything - just let whatever happen.
+    NotForcing,
+    /// Force a new era, then reset to `NotForcing` as soon as it is done.
+    /// Note that this will force to trigger an election until a new era is triggered, if the
+    /// election failed, the next session end will trigger a new election again, until success.
+    ForceNew,
+    /// Avoid a new era indefinitely.
+    ForceNone,
+    /// Force a new era at the end of all sessions indefinitely.
+    ForceAlways,
+}
+
+impl Default for Forcing {
+    fn default() -> Self {
+        Forcing::NotForcing
+    }
+}
+
+/// A record of the nominations made by a specific account.
+#[derive(PartialEq, Eq, Clone, Encode, Decode, RuntimeDebug)]
+pub struct Nominations<AccountId, Balance> {
+    /// The targets of nomination and amounts of staking.
+    pub targets: Vec<(AccountId, Balance)>,
+    /// The era the nominations were submitted.
+    ///
+    /// Except for initial nominations which are considered submitted at era 0.
+    pub submitted_in: EraIndex,
+    /// Whether the nominations have been suppressed.
+    pub suppressed: bool,
+}
+
+/// Reward points of an era. Used to split era total payout between dapps rewards.
+///
+/// This points will be used to reward contracts operators and their respective nominators.
+#[derive(PartialEq, Encode, Decode, Default, RuntimeDebug)]
+pub struct EraStakingPoints<AccountId: Ord, Balance: HasCompact> {
```

Comment:
A bit unclear how points will be calculated for dapps, is it dear code?

Reply:
It is from old codes. So, it may change the format.